### PR TITLE
model-per-category scripts implementing root node field selection

### PR DESCRIPTION
**Pull request by jaor**

### CODE REVIEW

File: model-per-category/batch-prediction/readme.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Batch predictions with category models
+
+This scripts uses the result of
```
Comment (akashenfelter) [inline model-per-category/batch-prediction/readme.md]: Typo: "This script uses the result of..."

**Comment by jaor** [comment]
(i've just rebased this PR on top of current master, after the recent merge)

File: model-per-category/readme.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## How to install
+
+### Using bigmler
+
+If you have bigmler installed in your system, just checkout this
+repository and, at its top level, issue the command:
+
+        make compile PKG_DIR=model-per-category
+
+That will create for you all three scripts.
```
Comment (akashenfelter) [inline model-per-category/readme.md]: Idiomatic change: "That will create all three scripts for you."

File: model-per-cluster/readme.md [hunk 3]
```diff
@@ -24,11 +24,23 @@ make predictions.
...
 
+### Using bigmler
+
+If you have bigmler installed in your system, just checkout this
+repository and, at its top level, issue the command:
+
+        make compile PKG_DIR=model-per-cluster
+
+That will create for you all necessary libraries and the three
+scripts.
```
Comment (akashenfelter) [inline model-per-cluster/readme.md]: Similar idiomatic change: "That will create all necessary libraries and the three scripts for you."

**Review by akashenfelter — CHANGES_REQUESTED** [review]
Just a couple typos/nitpicks.

File: model-per-category/create-category-models/script.whizzml [hunk 4]
```diff
@@ -0,0 +1,94 @@
+;; Given a dataset, a field and a model kind, splits the dataset
+;; per category and uses the results to creates models of the given
```
Comment (mmerce) [inline model-per-category/create-category-models/script.whizzml]: uses the results to create...

File: model-per-category/create-category-models/script.whizzml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    (when (not (= "categorical" (f "optype")))
+      (raise (str "Field '" field-id "' is not categorical")))
+    (let (cs (map head (f ["summary" "categories"] [])))
+      (when (empty? cs)
+        (raise (str "Field '" field-id "' does not contain any category")))
+      [cs (f ["summary" "missing_count"] 0) id (f "name")])))
+
+;; Given a parent dataset, a field and one of its categories,
+;; create a new dataset containing only the instances of the parent
+;; that have the given given category.
```
Comment (mmerce) [inline model-per-category/create-category-models/script.whizzml]: that have the given category

**Comment by jaor** [comment]
Documentation typos fixed in d373bc2.

**Pull request merged**